architecture/authentication.adoc

```diff
@@ -0,0 +1,268 @@
+= Authentication
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title: 
+
+toc::[]
+
+== Overview
+
+The authentication layer is responsible for identifying the user associated with requests to the OpenShift API. Information about the requesting user is then used by the authorization layer to determine whether the request should be allowed.
+
+== API Authentication
+
+Requests to the OpenShift API are authenticated using the following methods:
+
+* OAuth Access Token
+** Obtained from the OpenShift OAuth server using the `/oauth/authorize` and `/oauth/token` endpoints
+** Sent as an "Authorization: Bearer &hellip;" header or an "access_token=&hellip;" query parameter
+* X.509 Certificate
+** Requires a https connection to the API server
+** Verified by the API server against a trusted certificate authority bundle
+** The API server creates and distributes certificates to controllers to authenticate themselves
+
+Any request with an invalid access token or an invalid certificate is rejected by the authentication layer.
+
+If no access token or certificate is presented, the authentication layer assigns the `system:anonymous` user and the `system:unauthenticated` group to the request. This allows the authorization layer to determine which requests (if any) an anonymous user is allowed to make.
+
+== OAuth
+
+=== Overview
```

Comment:
Make it clear that oauth is secured differently than the API server and that all these bajillion parameters only affect parts under /oauth, not the API server.

It seems obvious, but it may seem weird to people that they can authenticate to oauth using their session, but then the api server won't respect it even though they "told openshift to use sessions to recognize users"